express_relay/sdk/python/express_relay_utils/express_relay_client.py

```diff
@@ -0,0 +1,404 @@
+
+    def process_response_msg(self, msg: dict) -> dict:
+        """
+        Processes a response message received from the server via websocket.
+
+        Args:
+            msg: The message to process.
+        Returns:
+            The result field of the message.
+        """
+        if msg.get("status") and msg.get("status") != "success":
+            raise ExpressRelayClientException(
+                f"Error in websocket response with message id {msg.get('id')}: {msg.get('result')}"
+            )
+        return msg["result"]
+
+    async def subscribe_chains(self, chain_ids: list[str]):
+        """
+        Subscribes websocket to a list of chain IDs for new opportunities.
+
+        Args:
+            chain_ids: A list of chain IDs to subscribe to.
+        """
+        params = {
+            "chain_ids": chain_ids,
+        }
+        await self.send_ws_message("subscribe", params)
+
+    async def unsubscribe_chains(self, chain_ids: list[str]):
+        """
+        Unsubscribes websocket from a list of chain IDs for new opportunities.
+
+        Args:
+            chain_ids: A list of chain IDs to unsubscribe from.
+        """
+        params = {
+            "chain_ids": chain_ids,
+        }
+        await self.send_ws_message("unsubscribe", params)
+
+    async def submit_bid(
+        self, bid: Bid, subscribe_to_updates: bool = True, **kwargs
+    ) -> UUID:
+        """
+        Submits a bid to the auction server.
+
+        Args:
+            bid: An object representing the bid to submit.
+            subscribe_to_updates: A boolean indicating whether to subscribe to the bid status updates.
+            kwargs: Keyword arguments to pass to the HTTP post request.
+        Returns:
+            The ID of the submitted bid.
+        """
+        bid_dict = bid.model_dump()
+        if subscribe_to_updates:
+            result = await self.send_ws_message("post_bid", bid_dict)
+            bid_id = UUID(result.get("id"))
+        else:
+            async with httpx.AsyncClient() as client:
+                resp = await client.post(
+                    urllib.parse.urlparse(self.server_url)
+                    ._replace(path="/v1/bids")
+                    .geturl(),
+                    json=bid_dict,
+                    **kwargs,
+                )
+
+            resp.raise_for_status()
+            bid_id = UUID(resp.json().get("id"))
+
+        return bid_id
+
+    async def submit_opportunity_bid(
+        self,
+        opportunity_bid: OpportunityBid,
+        subscribe_to_updates: bool = True,
+        **kwargs,
+    ) -> UUID:
+        """
+        Submits a bid on an opportunity to the server via websocket.
+
+        Args:
+            opportunity_bid: An object representing the bid to submit on an opportunity.
+            subscribe_to_updates: A boolean indicating whether to subscribe to the bid status updates.
+            kwargs: Keyword arguments to pass to the HTTP post request.
```

Comment:
this parameter is now a bit ill defined. If `subscribe_to_updates` is True, this is useless, but we don't communicate that with the users. Anyway, I think we should reuse a single httpx client and allow the user to pass in any extra configuration required in the constructor.